File: client/src/common/gameboard.ts
export interface GameBoard {
  id: string;
  wordList: string[];
  cards: GameCard[];
}

export interface GameCard {
  word: string;
  color: Color;
  hasBeenGuessed: boolean;
}

export enum Color {
  Red = 0,
  Blue = 1,
  Neutral = 2,
  Assassin = 3,
  Unknown = 4,
}

File: server/src/gameboard.go
package main

type GameBoard struct {
	ID    string                     `json:"id"`
	Cards *[wordListLength]*GameCard `json:"cards"`
}

type GameCard struct {
	Word           string `json:"word"`
	Color          Color  `json:"color"`
	HasBeenGuessed bool   `json:"hasBeenGuessed"`
}

type Color int

const (
	RED Color = iota
	BLUE
	NEUTRAL
	ASSASSIN
	UNKNOWN
)
